code/game/objects/items/grenades/grenade.dm

```diff
@@ -27,6 +27,8 @@
 	var/clumsy_check = GRENADE_CLUMSY_FUMBLE
 	///Was sticky tape used to make this sticky?
 	var/sticky = FALSE
+	///Is this grenade being used to suicide?
+	var/mob/living/carbon/suicider
```

Comment:
You need to set this to `null` after you're done using it and in `Destroy`, otherwise you will cause hard deletes.